### DISCUSSION TITLE
Enable GPG

### DISCUSSION THREAD

**Discussion by williamdes** [upptime/upptime#245] [Ideas] [answered]
I would really like to have GPG signed commits
Any way I could insert a custom action that would fill them in?
Or maybe you could support that natively

The project is awesome but this was the missing detail I closed my eyes on. Just hope that it can be resolved in some way :)

Goal: re-enable the rule for branch protection to require GPG signed commits.

**Comment by AnandChowdhary** [comment, marked as answer]
This is currently not trivial because we'll have to add GPG keys as secrets in the repository, import them to sign commit, and then make signed commits using the personal access token. Of course, @upptime-bot cannot be used; this will be limited to users using their own personal access tokens or custom bot accounts.

A nice way to do this is for now is https://github.com/actions/runner/issues/667#issuecomment-775277257, I'll be happy to accept a PR that parses GPG keys from secrets and signs commits.

**Reply by williamdes** [reply]
> A nice way to do this is for now is actions/runner#667 (comment), I'll be happy to accept a PR that parses GPG keys from secrets and signs commits.

Why not allowing the template to have a GPG step that would work only if the secret is defined ?
